CRM/Contact/BAO/Query.php

```diff
@@ -2130,7 +2130,7 @@ public function restWhere(&$values) {
 
     $setTables = TRUE;
 
-    $locationType = CRM_Core_PseudoConstant::get('CRM_Core_DAO_Address', 'location_type_id');
+    $locationType = CRM_Core_DAO_Address::buildOptions('location_type_id', 'validate');
     if (isset($locType[1]) && is_numeric($locType[1])) {
```

Comment:
The same thing could be achieved with:
`CRM_Core_PseudoConstant::get('CRM_Core_DAO_Address', 'location_type_id', [], 'validate');`

I'm not sure which  of `CRM_Core_PseudoConstant::get` and CRM_Core_DAO_Address::buildOptions` is preferred - @eileenmcnaughton ??

In this case, I think you could further simplify this to something like:
`CRM_Core_PseudoConstant::getName('CRM_Core_DAO_Address', 'location_type_id', $locType[1]);`

Reply:
Maybe `buildOptions` is not the best solution, but I chose it to be the same as the code used on line 2790.

Reply:
@mattwire I think its ok to use buildOptions in here as we are doing in other places too:
```
CRM//Contact/Selector.php:419:        $locationTypes = CRM_Core_DAO_Address::buildOptions('location_type_id', 'validate');
CRM//Contact/BAO/Query.php:1016:    $locationTypes = CRM_Core_DAO_Address::buildOptions('location_type_id', 'validate');
CRM//Contact/BAO/Query.php:2371:      $locationType = CRM_Core_DAO_Address::buildOptions('location_type_id', 'validate');
CRM//Contact/BAO/Query.php:2790:          $locationTypes = CRM_Core_DAO_Address::buildOptions('location_type_id', 'validate');
CRM//Core/BAO/Mapping.php:1132:    $locationTypes = CRM_Core_DAO_Address::buildOptions('location_type_id', 'validate');
```